### PR TITLE
Initial structure of pulpcore-plugin repo

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #4192
https://pulp.plan.io/issues/4192

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,19 +26,16 @@ script: source .travis/script.sh
 
 jobs:
   include:
-  - stage: deploy-pulpcore
-    script: bash .travis/deploy.sh pulpcore
-    if: tag =~ ^pulpcore-3.0*
   - stage: deploy-plugin
-    script: bash .travis/deploy.sh plugin
+    script: bash .travis/deploy.sh
```
Comment (dkliban) [inline .travis.yml]: we should change the deployment stage to be the same as for pulp_file.

File: .travis/script.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ fi
 flake8 --config flake8.cfg || exit 1
 
 # Run unit tests.
-coverage run manage.py test ./pulpcore/tests/unit/ ./plugin/tests/unit/
+coverage run manage.py test ./pulpcore/tests/unit/
```
Comment (dkliban) [inline .travis/script.sh]: shouldn't this be running the plugin tests?

**Comment by daviddavis** [comment]
The manage.py script is still needed because this repo uses django stuff like models and it needs to load the django environment.

File: .travis/script.sh (hunk 2 above)
Comment (dkliban) [reply .travis/script.sh]: cool.  makes sense.

**Comment by dkliban** [comment]
Since you are installing pulpcore anyway, I think you can use pulp-manager instead of manage.py.

File: .travis.yml [hunk 3]
```diff
@@ -26,19 +24,22 @@ script: source .travis/script.sh
...
-    script: bash .travis/deploy.sh plugin
+  - stage: deploy
+    script: skip
     if: tag =~ ^pulpcore-plugin*
+    deploy:
+      provider: pypi
+      distributions: sdist bdist_wheel
+      user: pulp
+      password:
+        secure: # TODO
```
Comment (daviddavis) [inline .travis.yml]: Created a subtask for this: https://pulp.plan.io/issues/4224

**Pull request merged**